uwblueprint/planet-read
# Implement submit translation flow

## Notion ticket link



[Implement submit translation flow](https://www.notion.so/uwblueprintexecs/Implement-Submit-Translation-flow-d4fd2b35dfc94bc6a009f05baaf4a187)



## Implementation description

- modified the primary button in review platform to serve returning to translator and submitting translation use cases
- extended auth requirements for updating stage



## Steps to test

1. log in as carl
2. send the great gatsby for review
3. log in as dwight
4. return the translation to translator. notice that everything is still working as intended
5. repeat steps 1-2
6. approve all lines. Notice that the primary button now says "submit translation"
7. click submit translation. play around with opening and closing the modal
8. confirm the modal
9. verify that you are redirected to the home page
10. verify that the translation is in PUBLISH stage



## What should reviewers focus on?

- does it work
- does the code make sense 

## Checklist
